stdpopsim/genomes.py

```diff
@@ -138,14 +198,96 @@ class Contig:
+        """
+        Default is a genetic map sized genomic element
+        """
+        return [GenomicElementType(start_positions=np.array(
+            [0]), end_positions=np.array([int(self.length)]))]
+
+    @mutation_types.default
+    def _get_default_mut_types(self):
+        # One neutral mutation type, simulated with msprime
+        return [stdpopsim.ext.MutationType()]
+
+    def __attrs_post_init__(self):
+        mut_ids = [
+            mtid for g in self.genomic_element_types for mtid in g.mutation_type_ids]
+        if set(range(len(self.mutation_types))) != set(mut_ids):
+            raise ValueError("The mutation type ids in the genomic elements "
+                             "must match the list of mutation types.")
+        neutral_mut_ids = [i for i, m in enumerate(
+            self.mutation_types) if m.distribution_type == "f" and m.distribution_args[0] == 0]
+        if any([g.proportions[g.mutation_type_ids.index(i)] != 0 for g in self.genomic_element_types for i in neutral_mut_ids]):
+            logger.warning(f"Warning: you defined neutral mutation type(s) "
+                           "which have non-zero proportions. This means "
+                           "neutral mutations will be simulated in SLiM, what "
+                           "can considerably slow down your simulations. If "
+                           "your neutral mutation type(s) are assigned "
+                           "proportions equal to zero, then they will be "
+                           "added onto the tree sequence using msprime.")
+        self.slim_fractions = [sum(g.proportions) for g in self.genomic_element_types]
+        # TODO: make sure genomic elements are non-overlapping within and across types
+
+    @property
+    def genomic_element_breaks(self):
+        breaks = np.concatenate([np.concatenate([g.start_positions, g.end_positions], axis=0) for g in self.genomic_element_types], axis=0)
+        breaks = np.sort(breaks)
+        # non-unique elements are possible because starts are inclusive but ends are not
+        breaks = np.unique(breaks)
+        if np.any(breaks > self.length):
+            raise ValueError("Genomic elements are outside the genomic region"
+                             " implied by the recombination map.")
+        if np.isclose(breaks[0], 0).any():
+            breaks = breaks[1:]
+        if not np.isclose(breaks[-1], int(self.length)).any():
+            breaks = np.concatenate([breaks, np.array([int(self.length)])], axis=0)
+        return breaks
+
+    @property
+    def slim_mutation_rate_map(self):
+        # 2D array with end positions
+        ends = np.array([g.end_positions for g in self.genomic_element_types])
+        # bool 2D (# genome element types x # breaks)
+        break_in_ends = (self.genomic_element_breaks==ends[...,None]).any(axis=1)
+        # array with getype index for each break
+        which_getype = np.apply_along_axis(_where_is_true, 0, break_in_ends)
```

Comment:
Maybe the `_where_is_true()` function could be moved up to here as a nested function, so that it's obvious that this is the only place the function is used. Unless there's a reason not to?